File: kotitehtavat/Description.java
public class Description {
    private String description;

    public Description(String description) {
        setDescription(description);
    }

    public void setDescription(String description) {
        int length;
        final int maxLength = 500;
        try {
            length = description.length();
        } catch (NullPointerException npe) {
            throw new IllegalArgumentException("Description can't be null"); 
        }
        length = description.length();
        if (length < 1) {
            throw new IllegalArgumentException("Description can't be empty");
        }
        if (length > 500) {
            throw new IllegalArgumentException("Description can't be over 500 characters long");
        }
        this.description = description;
    }

    public String getDescription() {
        return this.description;
    }

    @Override
    public String toString() {
        return description;
    }
}

File: kotitehtavat/visual-studio/tehtava14/Description.cs
public class Description
{
    public string Value { get; }

    public Description(string value)
    {
        if (string.IsNullOrEmpty(value))
        {
            throw new ArgumentException("Description cannot be null or empty");
        }
        if (value.Length > 500)
        {
            throw new ArgumentException("Description is too long");
        }
        Value = value;
    }

    public override string ToString()
    {
        return Value;
    }
}